ios/brave-ios/Sources/Brave/Frontend/Settings/General/SearchEngines/SearchQuickEnginesViewController.swift

```diff
@@ -170,7 +217,32 @@ extension SearchQuickEnginesViewController {
     if toggle.isOn {
       searchEngines.enableEngine(engine)
     } else {
-      searchEngines.disableEngine(engine, type: isPrivateBrowsing ? .privateMode : .standard)
+      searchEngines.disableEngine(
+        engine,
+        type: isPrivateBrowsing ? .privateMode : .standard
+      )
+    }
+    delegate?.searchQuickEnginesUpdated()
+  }
+
+  @objc func onEditButton() {
+    setEditing(!isEditing, animated: true)
+    editButton.image = isEditing ? nil : UIImage(systemName: "slider.horizontal.3")
+    editButton.title = isEditing ? Strings.done : nil
+  }
+
+  @objc func onAddButton() {
+    let addCustomSearchEngineVC = CustomEngineViewController(
+      profile: self.profile,
+      isPrivateBrowsing: self.isPrivateBrowsing
+    )
+    addCustomSearchEngineVC.onAddSucceed = { [weak self] in
+      self?.tableView.reloadData()
+      self?.delegate?.searchQuickEnginesUpdated()
     }
+    let navVC = UINavigationController(rootViewController: addCustomSearchEngineVC)
+    navVC.sheetPresentationController?.detents = [.large()]
```

Comment:
Whats the point of using `detents` if you're only allowing `large`/fullscreen presentation?